### ISSUE TITLE
Add pair and triplet samplers

### ISSUE THREAD

**Issue by monatis** [qdrant/quaterion#24]
## Why?
- Most loss functions work on pairs or triplets.
- Many datasets do not offer ready-to-use pairs and/or triplets, and they need forming on the fly.

## How?
- PyTorch has a built-in [`torch.utils.data.Sampler`](https://pytorch.org/docs/stable/_modules/torch/utils/data/sampler.html) class that can be passed to `DataLoader` alongside a `Dataset`. We may subclass it to form pairs and triplets.
